code/game/atoms_movable.dm

```diff
@@ -1088,7 +1094,10 @@
 	SHOULD_CALL_PARENT(TRUE)
 	if(status_flags & INCORPOREAL) //Incorporeal can normally pass through anything
 		blocker_opinion = TRUE
-
+	if(flags_pass & FLYING)	//An object marked as FLYING can cross over anything so long as it is not out of bounds, like
+		var/area/blocker_area = get_area(blocker)
+		if(blocker_area.outside && !isspacearea(blocker_area))
+			blocker_opinion = TRUE
```

Comment:
more snowflake, the more I look the more it feels like you less want flags and to just use canpassthrough